bokehjs/src/lib/api/palettes.ts

```diff
@@ -1011,6 +1012,39 @@ export const colorblind = {
   Colorblind,
 }
 
+export function interp_palette(palette: Color[], n: number): Color[] {
+  const npalette = palette.length
+  if (npalette < 1)
+    throw new Error("palette must contain at least one color")
+  if (n < 0)
+    throw new Error("requested palette length cannot be negative")
+
+  // Arrayable.interpolate operates on whole arrays, not slices, so need separate
+  // arrays for each of the R, G, B and A components.
+  const r = new Uint8Array(n)
+  const g = new Uint8Array(n)
+  const b = new Uint8Array(n)
+  const a = new Uint8Array(n)
+  for (let i = 0; i < npalette; i++) {
+    [r[i], g[i], b[i], a[i]] = color2rgba(palette[i])
+  }
+
+  const integers = range(0, npalette)
+  const fractions = linspace(0, npalette-1, n)
+
+  const r_interp = interpolate(fractions, integers, r)
+  const g_interp = interpolate(fractions, integers, g)
+  const b_interp = interpolate(fractions, integers, b)
+  const a_interp = interpolate(fractions, integers, a)
+
+  const ret = new Array<string>(n)
+  for (let i = 0; i < n; i++) {
+    ret[i] = color2hex(encode_rgba([r_interp[i], g_interp[i], b_interp[i], a_interp[i]]))
```

Comment:
What's the reason for generating hex string representations? wouldn't RGBA ints work here as well?

Reply:
Hex strings are what the Python implementation of the function return, hence I made this BokehJS equivalent return the same. Yes they could return any of the many other BokehJS color forms such as individual RGBA ints or encoded `uint32`.

Reply:
I think I might suggest RGBA ints as an alternative, since I think they will end up being more efficiently usable.

Reply:
OK, switching to returning RGBA integers.